### DISCUSSION TITLE
Should we emit the segment data in the event arguments in SablierV2Pro?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#21] [General]
It looks quite verbose when there are many segments.

<img width="1728" alt="Screen Shot 2022-06-07 at 5 07 48 PM" src="https://user-images.githubusercontent.com/8782666/172401415-a1d1c370-6cca-4139-b50d-099c24c7646e.png">

**Comment by PaulRBerg** [comment]
Yes I think we should keep emitting the full segment data in the event, since without it, it would be quite difficult (or impossible?) to reconstruct the pro stream just by listening to events.

**Reply by razgraf** [reply]
What are the gas implications for events with a big number of segments being emitted? For a stream with let's say 10-15 segments (or whatever the median size would be) does it increase the consumption by a meaningful amount?

Also, how are you looking to emit this information? Can you emit an actual array or is it going to be compressed into bytes/another primitive?

**Comment by PaulRBerg** [comment]
For future reference: what is being referred to as `SablierV2Pro` in this discussion is now called `SablierV2LockupDynamic`.

Ref https://github.com/sablier-labs/v2-core/discussions/371
